### PR TITLE
MessageLoop be gone

### PR DESCRIPTION
**Pull request by jellefoks**
`base::MessageLoop` is deprecated and removed.

The core of the change is to replace usage of "message loop" with
"task runner", to be more precise:

Replace `base::MessageLoop` references (from
`base/message_loop/message_loop.h`) with `base::SequencedTaskRunner`
(from `base/task/sequenced_task_runner.h`), and replace
`base::MessageLoop` instantiations with it's modern replacement which
depends on the use case.

Instead of getting a message loop with `base::MessageLoop::current()`,
get the task runner with
`base::SequencedTaskRunner::GetCurrentDefault()`.

Also don't use `base::ThreadTaskRunnerHandle::Get()` because that is
superseded by `base::SequencedTaskRunner::GetCurrentDefault()`.

The checking if you're in the right "thread", instead of checking the
current message loop against the stored (e.g.
`base::MessageLoop::current() == message_loop_)`), check the task
runner with `task_runner_->RunsTasksInCurrentSequence()`.

Don't use the more specific `task_runner->BelongsToCurrentThread()`
unless the task runner is known to need its own dedicated thread, to
make future movement of the task runner to the threadpool easier.
Prefer using `base::SequencedTaskRunner` over
`base::SingleThreadTaskRunner` for the same reason.

For `PostBlockingTask` and `WaitForFence`, don't directly use the
Starboard `TaskRunner` API customization anymore.  Use the equivalent
functions in `base::task_runner_util` from
`cobalt/base/task_runner_util.h` instead.

Instead of `base::MessageLoop::DestructionObserver`, use
`base::CurrentThread::DestructionObserver`.  Instead of calling
`message_loop->AddDestructionObserver()` from any thread, call
`base::CurrentThread::Get()->AddDestructionObserver()` from the
thread that needs to be observed for destruction.

In unit test scaffolding, when a task environment is needed, instead
of instantiating a `base::MessageLoop` object, instantiate a
`base::test::TaskEnvironment` (multiple threads) or
`base::test::SingleThreadTaskEnvironment` (main thread only) from
`base/test/task_environment.h` instead. Also, insteadof using
a `RunLoop::Run*()` method, use the equivalent from `TaskEnvironment`
instead.

When not in (unit) test framework, `base::MessageLoop` objects should
instead be `base::SingleThreadTaskExecutor` objects. Unlike a
`MessageLoop`, a `SingleThreadTaskExecutor` should not call `Start`,
as it is implicit.

Note: This CL includes removing `MessageLoop` references, but does
not yet include all `TaskEnvironment` instantiations that need to be
added to the scaffolding for the tests to be able to run, nor the
`RunLoop` change mentioned above.

b/326115779
b/326097526

### CODE REVIEW

File: cobalt/base/poller.h
Comment (andrewsavage1) [inline cobalt/base/poller.h:1]: Was Poller unused? Or do we explicitly not want to use it anymore for some reason?

Comment (jellefoks) [reply cobalt/base/poller.h:1]: It was totally unused.

File: cobalt/base/wrap_main.h [hunk 1]
```diff
@@ -68,68 +68,6 @@ void NoopStopFunction() {}
 
 #if defined(STARBOARD)
 #include "cobalt/base/wrap_main_starboard.h"
-#else
```
Comment (andrewsavage1) [inline cobalt/base/wrap_main.h:71]: Could you check if these are used by the native or host toolchains in trunk? I.e. if you make this same change (just for this file) in trunk, I assume we'd expect it to pass

Comment (jellefoks) [reply cobalt/base/wrap_main.h:71]: > Could you check if these are used by the native or host toolchains in trunk? I.e. if you make this same change (just for this file) in trunk, I assume we'd expect it to pass

I think it's unlikely to still be used but it's a good idea to give that a try and see of all the builds are green.

Comment (jellefoks) [reply cobalt/base/wrap_main.h:71]: Running at #2543

Comment (jellefoks) [reply cobalt/base/wrap_main.h:71]: It's all green. I guess that PR can just be merged. It shouldn't conflict with the same change here when merging back.

Comment (jellefoks) [reply cobalt/base/wrap_main.h:71]: After adding on_device, there are some failures, I'll investigate.

**Review by andrewsavage1 — APPROVED** [review]
All looks good to me, thanks Jelle!

**Pull request merged**